test/unit/url.cpp

```diff
@@ -640,19 +640,21 @@ class url_test
     //--------------------------------------------
 
     void
-    testHost()
+    set(string_view s1,
```

Comment:
As with other PRs. This fixes warnings related to default parameters in lambdas.

The warning:

```
warning: default argument specified for lambda parameter [-pedantic]
```

The reason is in Section 5.1.2 paragraph 5.

> Default arguments (8.3.6) shall not be specified in the parameter-declaration-clause of a lambda-declarator.